### PR TITLE
Profile delete

### PR DESCRIPTION
**Pull request by TarasOrlovskyi**

### CODE REVIEW

File: src/main/java/com/vinylteam/vinyl/web/servlets/EditProfileServlet.java [hunk 1]
```diff
@@ -66,7 +66,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
                 } else {
                     boolean checkOldPassword = securityService.checkPasswordAgainstUserPassword(user, oldPassword.toCharArray());
                     if (checkOldPassword) {
-                        boolean isEdit = userService.update(email, newEmail, newPassword, newDiscogsUserName);
+                        boolean isEdit;
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/web/servlets/EditProfileServlet.java]: isUpdated.

File: src/main/java/com/vinylteam/vinyl/web/servlets/EditProfileServlet.java [hunk 2]
```diff
@@ -66,7 +66,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+                        if (!newPassword.equals("")){
+                            isEdit = userService.update(email, newEmail, newPassword, newDiscogsUserName);
+                        } else {
+                            isEdit = userService.update(email, newEmail, oldPassword, newDiscogsUserName);
+                        }
                         logger.debug("Got result of edit user by email with " +
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/web/servlets/EditProfileServlet.java]: It's better to put logger in if - else:
logger.info("User was updated with new email or password in the database {'newEmail':{}}.", newEmail);

and

logger.info("Failed to update with new email or password user in the database {'newEmail':{}}.", newEmail);

File: src/main/java/com/vinylteam/vinyl/web/servlets/SignUpServlet.java [hunk 3]
```diff
@@ -49,24 +49,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         String discogsUserName = request.getParameter("discogsUserName");
         attributes.put("email", email);
         attributes.put("discogsUserName", discogsUserName);
-        if (!password.equals(confirmPassword)) {
+        if (password.equals("")){
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/web/servlets/SignUpServlet.java]: Add same checks to front on profile as you have on /signUp: password at least 8 characters, uppercase, lowercase, number.

File: src/main/java/com/vinylteam/vinyl/Starter.java [hunk 4]
```diff
@@ -102,6 +102,7 @@ public void run() {
         SignOutServlet signOutServlet = new SignOutServlet();
         ProfileServlet profileServlet = new ProfileServlet();
         EditProfileServlet editProfileServlet = new EditProfileServlet(securityService, userService);
+        DeleteUserServlet deleteUserServlet = new DeleteUserServlet(userService);
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/Starter.java]: Please rename to deleteProfileServlet

File: src/main/resources/templates/editProfile.html [hunk 5]
```diff
@@ -110,7 +110,7 @@
             <h2>Edit Profile</h2>
           </div>
 
-          <form action="/editProfile" method="POST" class="user edit-profile-user" name="form_edit" OnSubmit="return matchPassword();">
+          <form action="/editProfile" id="editProfileForm" method="POST" class="user edit-profile-user" name="form_edit" OnSubmit="return matchPassword();">
```
Comment (Greengrer) [inline src/main/resources/templates/editProfile.html]: id="edit-profile-form"

File: src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java [hunk 6]
```diff
@@ -65,6 +66,31 @@ public boolean add(User user) {
         return isAdded;
     }
 
+    @Override
+    public boolean remove(User user){
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java]: Format code please

**Review by Greengrer — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/web/servlets/SignUpServlet.java (hunk 3 above)
Comment (TarasOrlovskyi) [reply src/main/java/com/vinylteam/vinyl/web/servlets/SignUpServlet.java]: we have checks in editProfile and registration. I think that it's enough

File: src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java [hunk 7]
```diff
@@ -24,6 +24,7 @@
     private static final String INSERT = "INSERT INTO public.users" +
             " (email, password, salt, iterations, role, status, discogs_user_name)" +
             " VALUES (?, ?, ?, ?, ?, ?, ?)";
+    private static final String REMOVE = "DELETE FROM users WHERE email = ?";
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java]: Maybe rename to DELETE?

File: src/main/java/com/vinylteam/vinyl/dao/UserDao.java [hunk 8]
```diff
@@ -8,6 +8,8 @@
 
     boolean add(User user);
 
+    boolean remove(User user);
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/dao/UserDao.java]: Maybe rename to DELETE?

File: src/main/java/com/vinylteam/vinyl/service/UserService.java [hunk 9]
```diff
@@ -10,6 +10,8 @@
 
     boolean update(String oldEmail, String newEmail, String newPassword, String discogsUserName);
 
+    boolean remove(User user);
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/service/UserService.java]: Maybe rename to DELETE?

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java [hunk 10]
```diff
@@ -35,6 +35,11 @@ public boolean add(String email, String password, String discogsUserName) {
         return isAdded;
     }
 
+    @Override
+    public boolean remove(User user){
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java]: Maybe rename to DELETE?

File: src/main/java/com/vinylteam/vinyl/web/servlets/CatalogueServlet.java [hunk 11]
```diff
@@ -55,5 +55,4 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
             PageGenerator.getInstance().process("catalog", randomUniqueVinyls, attributes, response.getWriter());
         }
     }
-
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/web/servlets/CatalogueServlet.java:58]: Please live empty row before closing class

File: src/main/java/com/vinylteam/vinyl/web/servlets/DeleteUserServlet.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+import jakarta.servlet.http.HttpSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DeleteUserServlet extends HttpServlet {
+    private Logger logger = LoggerFactory.getLogger(getClass());
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/web/servlets/DeleteUserServlet.java]: You can just use annotation @Slf4j on the top of the class

File: src/main/java/com/vinylteam/vinyl/web/servlets/SignUpServlet.java (hunk 3 above)
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/web/servlets/SignUpServlet.java]: Please add formatting.

File: src/main/resources/templates/stores.html [hunk 13]
```diff
@@ -65,6 +86,8 @@
 
       </div>
 
+      <!-- #### -->
```
Comment (AlexKievDeveloper) [inline src/main/resources/templates/stores.html]: You really need this comment?

**Review by AlexKievDeveloper — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/web/servlets/CatalogueServlet.java (hunk 11 above)
Comment (TarasOrlovskyi) [reply src/main/java/com/vinylteam/vinyl/web/servlets/CatalogueServlet.java:58]: why?

Comment (AlexKievDeveloper) [reply src/main/java/com/vinylteam/vinyl/web/servlets/CatalogueServlet.java:58]: Code style

**Review by AlexKievDeveloper — APPROVED** [review]

File: src/main/java/com/vinylteam/vinyl/web/servlets/SignUpServlet.java [hunk 14]
```diff
@@ -49,24 +48,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
-            } else {
+            if (!password.equals(confirmPassword)) {
                 response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                logger.debug("Set response status to {'status':{}}", HttpServletResponse.SC_BAD_REQUEST);
-                attributes.put("message", "Sorry, but the user couldn't be registered. Check email, password or discogs username!");
+                log.debug("Set response status to {'status':{}}", HttpServletResponse.SC_BAD_REQUEST);
+                attributes.put("message", "Sorry, the passwords don't match!");
+            } else {
+                boolean isAdded = userService.add(email, password, discogsUserName);
+                log.debug("Got result of adding user with " +
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/web/servlets/SignUpServlet.java:62]: You can put this log in if-else as well.

**Review by Greengrer — APPROVED** [review]

**Pull request merged**